letsencrypt-compatibility-test/letsencrypt_compatibility_test/test_driver.py

```diff
@@ -0,0 +1,324 @@
+"""Tests Let's Encrypt plugins against different server configurations."""
+import argparse
+import filecmp
+import functools
+import logging
+import os
+import shutil
+import tempfile
+
+import OpenSSL
+
+from acme import challenges
+from acme import crypto_util
+from acme import messages
+from letsencrypt import achallenges
+from letsencrypt import errors as le_errors
+from letsencrypt import validator
+from letsencrypt.tests import acme_util
+
+from letsencrypt_compatibility_test import errors
+from letsencrypt_compatibility_test import util
+from letsencrypt_compatibility_test.configurators.apache import apache24
+
+
+DESCRIPTION = """
+Tests Let's Encrypt plugins against different server configuratons. It is
+assumed that Docker is already installed. If no test types is specified, all
+tests that the plugin supports are performed.
+
+"""
+
+PLUGINS = {"apache" : apache24.Proxy}
+
+
+logger = logging.getLogger(__name__)
+
+
+def test_authenticator(plugin, config, temp_dir):
+    """Tests authenticator, returning True if the tests are successful"""
+    backup = _create_backup(config, temp_dir)
+
+    achalls = _create_achalls(plugin)
+    if not achalls:
+        # Plugin/tests support no common challenge types
+        return True
```

Comment:
My natural inclination is that this could lead to some bugs that aren't recognized.  Maybe raise an exception?